scripts/generate_dump

```diff
@@ -144,21 +155,24 @@ save_bcmcmd_all_ns() {
 #  None
 ###############################################################################
 save_cmd() {
+    trap 'handle_error $? $LINENO' ERR
     local start_t=$(date +%s%3N)
     local end_t=0
     local cmd="$1"
     local filename=$2
     local filepath="${LOGDIR}/$filename"
     local do_gzip=${3:-false}
-    local save_stderr=${4:-true}
+    local save_stderr=${4:-$SAVE_STDERR}
```

Comment:
>save_stderr [](http://example.com/codeflow?start=10&length=11)

The intention of `save_stderr` is different from your new feature `-r`.

Actually I don't see the point of `save_stderr=False`.

Suggest you
1. remove old feature, always save stderr
2. add a new option like `inherit_stderr`, and still keep save stderr.

Reply:
The usecase of stderr=False as i see is only for the sonic-mgmt auto-techsupport test. 

If this is set to false, any intermediate stderr will be redirected to the stderr of the techsupport and the test can be enhanced to capture the stderr. Thereby we can have a single location to view the errors reported by any of the intermediate steps.

Reply:
> The usecase of stderr=False as i see is only for the sonic-mgmt auto-techsupport test.

Could you give a link of the usage in sonic-mgmt repo?

Reply:
No, it's not there yet. But we plan to update it once this gets merged.